### PR TITLE
Added support of ints

### PR DESCRIPTION
**Pull request by mintyowl**
In addition to uint, now indexer can index on ints (int64, int32 .. etc)

### CODE REVIEW

**Comment by luedigernet** [comment]
I'll appreciate if that would be merged.

Is there any chance that one of the maintainers will merge this? I've looked for memdb and want to use it for testing our codebase without the need to connect to a real database.  After a few steps i realised that at least the option for indexing Int fields is missing. Did I moss something if I say simple double the UintFieldIndex type as IntFieldIndex with all methods on it and replace all the uint types with the related int types. I've done so locally and it seems to work.

**Comment by HasMatthew** [comment]
Indexing on int types would be immensely helpful! 👍  to reviewing & merging this ASAP!

File: index.go [hunk 1]
```diff
@@ -250,6 +250,79 @@ func (s *StringMapFieldIndex) FromArgs(args ...interface{}) ([]byte, error) {
...
+	val := v.Int()
+	binary.PutUvarint(buf, uint64(val))
+
+	return buf, nil
+}
+
+// IsIntType returns whether the passed type is a type of Int and the number
+// of bytes the type is. To avoid platform specific sizes, the uint type returns
+// 8 bytes regardless of if it is smaller.
+func IsIntType(k reflect.Kind) (size int, okay bool) {
```
Comment (HasMatthew) [inline index.go]: `IsIntType` probably doesn't need to be exported, although I see the similar `IsUintType` is exported.

**Comment by HasMatthew** [comment]
@slackpad can you take a look at this? Thanks!

**Comment by jefferai** [comment]
@mintyowl Any chance you can sign the CLA? https://cla.hashicorp.com/hashicorp/go-memdb?pullRequest=29

File: index.go [hunk 2]
```diff
@@ -250,6 +250,79 @@ func (s *StringMapFieldIndex) FromArgs(args ...interface{}) ([]byte, error) {
...
+	switch k {
+	case reflect.Int:
```
Comment (jefferai) [inline index.go]: Can you use `binary.MaxVarint*` similar to the Uint code?

File: index.go [hunk 3]
```diff
@@ -250,6 +250,79 @@ func (s *StringMapFieldIndex) FromArgs(args ...interface{}) ([]byte, error) {
...
+		return 8, true
+	case reflect.Int8:
```
Comment (jefferai) [inline index.go]: Interestingly, the Uint case uses a size of 2 for this, not 1...any idea why?

Comment (jefferai) [reply index.go]: Ah, it's related to https://github.com/hashicorp/go-memdb/commit/4b10f0c25f7c6094cabf081f5d700c4e40c9365a#diff-a64dd4f91a0b1fb420c9e63773bfa8bf and https://github.com/hashicorp/go-memdb/issues/30 -- seems like size 2 is best for int8.

**Pull request closed without merging**